providers/common/der/der_dsa.c.in

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/bn.h>
+#include <openssl/obj_mac.h>
+#include "prov/der_dsa.h"
+
+/* Well known OIDs precompiled */
+{-
+    $OUT = oids_to_c::process_leaves('providers/common/der/DSA.asn1',
+                                     { dir => $config{sourcedir},
+                                       filter => \&oids_to_c::filter_to_C });
+-}
+
+size_t DER_w_algorithmIdentifier_DSA(unsigned char **pp,
+                                     const unsigned char *start,
+                                     DSA *dsa)
+{
+    size_t a = 0;
+
+    /* No parameters */
+    CHECKED(a, DER_w_precompiled(pp, start, der_oid_id_dsa,
+                                 sizeof(der_oid_id_dsa)));
+    CHECKED(a, DER_w_sequence(pp, start, a));
+    return a;
+}
+
+#define MD_CASE(name)                                                   \
```

Comment:
Somewhat philosophically, why have a macro to expand into blocks of code when the whole file is being generated?

Reply:
Not sure what that would look like...